geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/Histogram.scala

```diff
@@ -205,6 +207,17 @@ object Histogram {
     buf.asInstanceOf[(T, T)]
   }
 
+  /**
+   * Checks if 2 values map to the same bin
+   *
+   * @param o1 object 1
+   * @param o2 object 2
+   * @param bins number of binds
+   * @tparam T type binding
+   * @return
+   */
+  def equivalent[T: ClassTag](o1: T, o2: T, bins: Int): Boolean = Try(Binning(bins, (o1, o2))).isFailure
```

Comment:
Ok....  *blinks*  What is this doing?

Binning can fail for multiple reasons, right?  Does this definitely check equivalence?

Reply:
ah, i think the only way it would fail is if the endpoints end up mapping to the same bin. this is admittedly not the most elegant solution... but because you have to handle each binding differently, it's the easiest. previously we just compared the endpoints, but that doesn't handle the case where they are different but map to the same bin (which was the underlying bug).

Reply:
the rest of the changes are to separate out the logic for this with actually allocating the binned array, so that we're not allocating an array here.

Reply:
```suggestion
  def equivalent[T: ClassTag](o1: T, o2: T, bins: Int): Boolean =
     Try(Binning(bins, (o1, o2))).isFailure // this will fail if the two values map to the same bin
```